modules/imgcodecs/src/grfmt_png.cpp

```diff
@@ -732,7 +739,7 @@ uint32_t PngDecoder::read_chunk(Chunk& chunk)
     chunk.p.resize(size);
     memcpy(chunk.p.data(), size_id, 8);
     if (readFromStreamOrBuffer(&chunk.p[8], chunk.p.size() - 8))
-        return id;
+        return isBigEndian() ? id : *((uint32_t*)(&size_id[4]));
```

Comment:
since we just made `id` correct in both cases, why change the return here? wouldn't just `return id;` now be ok for both?